### PR TITLE
2015 01 06b

### PR DESCRIPTION
**Pull request by KamilWo**
Po poprawkach zgodnie z uwagami.

### CODE REVIEW

File: addons/bestja_frontend_fpbz/data/images.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+    <data noupdate="1">
+        <record id="image_bg_blue" model="ir.attachment">
+            <field name="name">back_enable.jpg</field>
+            <field name="datas_fname">back_enable.jpg</field>
+            <field name="res_model">ir.ui.view</field>
+            <field name="type">url</field>
+            <field name="url">/web/static/src/img/back_enable.jpg</field>
```
Comment (ludwiktrammer) [inline addons/bestja_frontend_fpbz/data/images.xml]: Does the `/web/static/src/img/back_enable.jpg` exists? I can't find it. What is the purpose of this code?

File: addons/bestja_frontend_fpbz/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+    # Theme information
+    'name': "FPBŻ frontend",
+    'description': """
+    This is custom theme made for Federacja Polskich Banków Żywności
```
Comment (ludwiktrammer) [inline addons/bestja_frontend_fpbz/__openerp__.py]: "A custom theme..."

**Comment by KrzysiekJ** [comment]
Runbot reports some errors:
```
2015-01-06 12:21:49,282 18445 ERROR 00356-84-e44784-fpbz openerp.addons.base.ir.ir_ui_view: Element '<xpath expr="//footer">' cannot be located in parent view

Error context:
View `Main layout`
[view_id: 388, xml_id: website.layout, model: n/a, parent_id: n/a]
2015-01-06 12:21:49,304 18445 CRITICAL 00356-84-e44784-fpbz openerp.service.server: Failed to initialize database `00356-84-e44784-fpbz`.
Traceback (most recent call last):
  File "/home/app/odoo-extra/runbot/static/build/00356-84-e44784/openerp/service/server.py", line 901, in preload_registries
    registry = RegistryManager.new(dbname, update_module=update_module)
  File "/home/app/odoo-extra/runbot/static/build/00356-84-e44784/openerp/modules/registry.py", line 340, in new
    openerp.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/home/app/odoo-extra/runbot/static/build/00356-84-e44784/openerp/modules/loading.py", line 358, in load_modules
    loaded_modules, update_module)
  File "/home/app/odoo-extra/runbot/static/build/00356-84-e44784/openerp/modules/loading.py", line 263, in load_marked_modules
    loaded, processed = load_module_graph(cr, graph, progressdict, report=report, skip_modules=loaded_modules, perform_checks=perform_checks)
  File "/home/app/odoo-extra/runbot/static/build/00356-84-e44784/openerp/modules/loading.py", line 182, in load_module_graph
    _load_data(cr, module_name, idref, mode, kind='data')
  File "/home/app/odoo-extra/runbot/static/build/00356-84-e44784/openerp/modules/loading.py", line 118, in _load_data
    tools.convert_file(cr, module_name, filename, idref, mode, noupdate, kind, report)
  File "/home/app/odoo-extra/runbot/static/build/00356-84-e44784/openerp/tools/convert.py", line 899, in convert_file
    convert_xml_import(cr, module, fp, idref, mode, noupdate, report)
  File "/home/app/odoo-extra/runbot/static/build/00356-84-e44784/openerp/tools/convert.py", line 985, in convert_xml_import
    obj.parse(doc.getroot(), mode=mode)
  File "/home/app/odoo-extra/runbot/static/build/00356-84-e44784/openerp/tools/convert.py", line 851, in parse
    self._tags[rec.tag](self.cr, rec, n, mode=mode)
  File "/home/app/odoo-extra/runbot/static/build/00356-84-e44784/openerp/tools/convert.py", line 827, in _tag_template
    return self._tag_record(cr, record, data_node)
  File "/home/app/odoo-extra/runbot/static/build/00356-84-e44784/openerp/tools/convert.py", line 765, in _tag_record
    id = self.pool['ir.model.data']._update(cr, self.uid, rec_model, self.module, res, rec_id or False, not self.isnoupdate(data_node), noupdate=self.isnoupdate(data_node), mode=self.mode, context=rec_context )
  File "/home/app/odoo-extra/runbot/static/build/00356-84-e44784/openerp/api.py", line 237, in wrapper
    return old_api(self, *args, **kwargs)
  File "/home/app/odoo-extra/runbot/static/build/00356-84-e44784/openerp/addons/base/ir/ir_model.py", line 1052, in _update
    res_id = model_obj.create(cr, uid, values, context=context)
  File "/home/app/odoo-extra/runbot/static/build/00356-84-e44784/openerp/api.py", line 237, in wrapper
    return old_api(self, *args, **kwargs)
  File "/home/app/odoo-extra/runbot/static/build/00356-84-e44784/openerp/addons/base/ir/ir_ui_view.py", line 264, in create
    context=context)
  File "/home/app/odoo-extra/runbot/static/build/00356-84-e44784/openerp/api.py", line 237, in wrapper
    return old_api(self, *args, **kwargs)
  File "/home/app/odoo-extra/runbot/static/build/00356-84-e44784/openerp/api.py", line 332, in old_api
    result = method(recs, *args, **kwargs)
  File "/home/app/odoo-extra/runbot/static/build/00356-84-e44784/openerp/models.py", line 3968, in create
    record = self.browse(self._create(old_vals))
  File "/home/app/odoo-extra/runbot/static/build/00356-84-e44784/openerp/api.py", line 235, in wrapper
    return new_api(self, *args, **kwargs)
  File "/home/app/odoo-extra/runbot/static/build/00356-84-e44784/openerp/api.py", line 464, in new_api
    result = method(self._model, cr, uid, *args, **kwargs)
  File "/home/app/odoo-extra/runbot/static/build/00356-84-e44784/openerp/models.py", line 4159, in _create
    recs._validate_fields(vals)
  File "/home/app/odoo-extra/runbot/static/build/00356-84-e44784/openerp/api.py", line 235, in wrapper
    return new_api(self, *args, **kwargs)
  File "/home/app/odoo-extra/runbot/static/build/00356-84-e44784/openerp/models.py", line 1300, in _validate_fields
    raise ValidationError('\n'.join(errors))
ParseError: "ValidateError
Field(s) `arch` failed against a constraint: Invalid view definition

Error details:
Element '<xpath expr="//footer">' cannot be located in parent view

Error context:
View `Main layout`
[view_id: 388, xml_id: website.layout, model: n/a, parent_id: n/a]" while parsing None:5, near
<data inherit_id="website.layout" primary="True">
    <!-- Add report attributes -->
    <xpath expr="//html" position="attributes">
        <attribute name="t-att-data-report-margin-top">data_report_margin_top if data_report_margin_top else None</attribute>
        <attribute name="t-att-data-report-header-spacing">data_report_header_spacing if data_report_header_spacing else None</attribute>
        <attribute name="t-att-data-report-dpi">data_report_dpi if data_report_dpi else None</attribute>
    </xpath>
    <!-- Add report style -->
    <xpath expr="//head" position="inside">
        <style type="text/css">
            <t t-call="report.style"/>
        </style>
    </xpath>
    <xpath expr="//body" position="attributes">
        <attribute name="class">container</attribute>
    </xpath>
    <xpath expr="//header" position="replace"/>
    <xpath expr="//footer" position="replace">
        <ul class="list-inline js_language_selector mt16" t-if="(request and request.website_multilang and len(languages) &gt; 1) or (website and editable)">
            <li t-foreach="languages" t-as="lg">
                <a t-att-href="url_for(request.httprequest.path + '?' + keep_query(), lang=lg[0])" t-att-data-default-lang="editable and 'true' if website and lg[0] == website.default_lang_code else None">
                    <t t-esc="lg[1].split('/').pop()"/>
                </a>
            </li>
            <li groups="base.group_website_publisher">
                <t t-set="url_return" t-value="url_for('', '[lang]') + '?' + keep_query()"/>
                <a t-attf-href="/web#action=base.action_view_base_language_install&amp;website_id=#{website.id if website else ''}&amp;url_return=#{url_return}">
                    <i class="fa fa-plus-circle"/>
                    Add a language...
                </a>
            </li>
        </ul>
    </xpath>
<xpath position="attributes" expr="."><attribute name="t-name">website_report.layout</attribute></xpath></data>
```

File: addons/bestja_frontend_fpbz/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        'views/config.xml',
+        'views/pages.xml',
+        'views/snippets.xml',
+        'data/images.xml',
+    ],
+    'application': True,
+    # Your information
+    'author': "Laboratorium EE, Kamil Woźniak",
+    'website': "http://laboratorium.ee",
+}
```
Comment (KrzysiekJ) [inline addons/bestja_frontend_fpbz/__openerp__.py]: This red sign indicates that there is no newline at end of file. It is good practice to put such newlines (however it is not critically important).

File: addons/bestja_frontend_fpbz/views/pages.xml [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+                                </li>
+                            </ul>
+                        </div>
+                        <div class="col-footer text-center">
+                            <h4><a href="http://www.wolontariat.bankizywnosci.pl/">O nas</a>
+                                <a href="http://www.wolontariat.bankizywnosci.pl/">Kontakt</a></h4>
+                        </div>
+                        <div class="col-footer text-right copyrights">
+                            <div>
+                                <p style="color: #FF9900">© 2014, Federacja Banków Żywności</p>
```
Comment (ludwiktrammer) [inline addons/bestja_frontend_fpbz/views/pages.xml]: Is there a reason for an inline CSS here?

**Pull request closed without merging**